src/MessagingBubble/NormalMessagingBubble.tsx

```diff
@@ -178,20 +185,29 @@ function _renderNonInteractiveTimestamp({
 function _renderActionButton({
   timestamp,
   onClickDeleteButton,
+  messageBody,
   metadataClassNames,
   actionButtonClassNames,
 }: {
   timestamp: Date;
   onClickDeleteButton: () => void;
+  messageBody: string;
   metadataClassNames: string;
   actionButtonClassNames: string;
 }): React.ReactNode {
+  const formattedTimestamp = _formatDateForTimestamp(timestamp);
+
   return (
     <div className={metadataClassNames}>
-      <Button className={actionButtonClassNames} type="linkPlain" onClick={onClickDeleteButton}>
+      <Button
+        ariaLabel={`Delete ${formattedTimestamp} ${messageBody || ""}`}
```

Comment:
The `|| ""` avoids having the screen-reader read out "undefined" if the prop isn't in for whatever reason.